db/migrate/20150713143226_add_unique_to_operatingsystems_title.rb

```diff
@@ -0,0 +1,17 @@
+class AddUniqueToOperatingsystemsTitle < ActiveRecord::Migration
+  def up
+    duplicities = Operatingsystem.all.group_by(&:title).values.keep_if { |ary| ary.length > 1}
+    duplicities.each do |systems|
+      systems.each do |os|
+        os.description = os.fullname
```

Comment:
race conditions within the app can cause it.  Katello creates an operating system (and install media) if it does not already exist after a repo is synced.  If you are syncing multiple repositories with the same medium at the same time, you can hit this issue (and i have seen it hit many times). 

I've tried to work around it in app code (https://github.com/Katello/katello/pull/5269), but i've even seen cases where they are both created but later updates fail (as the validation is then hit).   That caused me to conclude that a db validation is truly needed to solve the issue.